Code/src/main/java/nl/utwente/viskell/ui/components/ConnectionAnchor.java

```diff
@@ -25,6 +32,114 @@
  */
 public abstract class ConnectionAnchor extends StackPane implements ComponentLoader {
 
+    /**
+     * Handler class that reacts to user inputs on ConnectionAnchors to be able to
+     * create, edit and drag Connections.
+     */
+    private class AnchorHandler implements EventHandler<InputEvent> {
```

Comment:
Does this have to be a nested class? Can't you just pull its methods into the ConnectionAnchor class?